geonode/templates/search/_region_filter.html

```diff
@@ -0,0 +1,18 @@
+{% load i18n %}
+{% load base_tags %}
+<nav class="filter">
+
+    <h4><a href="#" class="toggle toggle-nav"><i class="fa fa-chevron-right"></i>{% trans "Regions" %}</a></h4>
+    <ul class="nav closed" id="regions">
+        <li>
+            <div class="input-group">
+            <input name="region_search_input" id="region_search_input" ng-model="text_query" type="text"
```

Comment:
ng-model="text_query" is the reason why the text appears in both places. You have to remove this ng-model.